src/utils.rs

```diff
@@ -8,8 +8,16 @@ pub(crate) const BASE64: engine::GeneralPurpose = engine::GeneralPurpose::new(
         .with_encode_padding(true),
 );
 
-pub(crate) fn is_tchar(c: u8) -> bool {
+fn is_tchar(c: u8) -> bool {
     // See tchar values list in https://tools.ietf.org/html/rfc7230#section-3.2.6
     let tchars = b"!#$%&'*+-.^_`|~";
     tchars.contains(&c) || c.is_ascii_alphanumeric()
 }
+
+pub(crate) fn is_token_start(c: u8) -> bool {
+    c.is_ascii_alphabetic() || c == b'*'
+}
+
+pub(crate) fn is_token_continue(c: u8) -> bool {
```

Comment:
Not a big fan of this name.
Maybe is_allowed_token_character? Or something like that?
Otherwise I'm not sure we necessarily need the helper, as it makes more difficult to follow the spec against the implementation `If the first character of input_string is not in tchar, ":", or "/", return output_string.` 🤷

Reply:
To me, that name implies that the character is valid anywhere in the token, but the first character is only a subset of those.

IMO exactly matching the spec is nice organization is nice, but centralizing the validation is better for code maintenance, especially as I intend to add a `Token` type as part of #98.

Reply:
Fair enough. Let's give them some better names like `is_allowed_inner_token_char` and `is_allowed_token_start_char`.

Reply:
Done.